modules/charon-core/src/test/java/org/wso2/charon3/core/protocol/endpoints/UserResourceManagerTest.java

```diff
@@ -162,42 +504,797 @@ public void testListWithGetInt(String filter, int startIndexInt, int countInt,
+        Assert.assertEquals(scimResponse.getResponseStatus(), expectedScimResponseStatus);
+    }
+
+    @DataProvider(name = "dataForTestCreatedUserResourceIsNull")
+    public Object[][] dataToTestCreatedUserResourceIsNull() {
+
+        return new Object[][]{
+                {NEW_USER_SCIM_OBJECT_STRING, "userName", null, ResponseCodeConstants.CODE_INTERNAL_ERROR}
+        };
+    }
+
+    @Test(dataProvider = "dataForTestCreatedUserResourceIsNull")
+    public void testCreatedUserResourceIsNull(String scimObjectString, String attributes,
+                                              String excludeAttributes, int expectedScimResponseStatus)
+            throws ConflictException, BadRequestException, CharonException {
+
+        abstractResourceManager.when(() -> AbstractResourceManager.getResourceEndpointURL(SCIMConstants.USER_ENDPOINT))
+                .thenReturn(SCIM2_USER_ENDPOINT);
+        abstractResourceManager.when(()
+                -> AbstractResourceManager.encodeSCIMException(any(InternalErrorException.class)))
+                .thenReturn(getEncodeSCIMExceptionObject(new InternalErrorException()));
+        Mockito.when(userManager.createUser(any(User.class), anyMap())).thenReturn(null);
+
+        SCIMResponse scimResponse = userResourceManager.create(scimObjectString,
+                userManager, attributes, excludeAttributes);
+        Assert.assertEquals(scimResponse.getResponseStatus(), expectedScimResponseStatus);
+
+    }
+
+    @DataProvider(name = "dataForTestCreateBadRequestException")
+    public Object[][] dataToTestCreatBadRequestException() {
+
+        return new Object[][]{
+                {NEW_USER_SCIM_OBJECT_STRING, "userName", null, ResponseCodeConstants.CODE_BAD_REQUEST}
+        };
+    }
+
+    @Test(dataProvider = "dataForTestCreateBadRequestException")
+    public void testCreateBadRequestException(String scimObjectString, String attributes,
+                                              String excludeAttributes, int expectedScimResponseStatus)
+            throws ConflictException, BadRequestException, CharonException {
+
+        abstractResourceManager.when(() -> AbstractResourceManager.getResourceEndpointURL(SCIMConstants.USER_ENDPOINT))
+                .thenReturn(SCIM2_USER_ENDPOINT);
+        abstractResourceManager.when(() -> AbstractResourceManager.encodeSCIMException(any(BadRequestException.class)))
+                .thenReturn(getEncodeSCIMExceptionObject(new BadRequestException()));
+        Mockito.when(userManager.createUser(any(User.class), anyMap())).thenThrow(BadRequestException.class);
+
+        SCIMResponse scimResponse = userResourceManager.create(scimObjectString,
+                userManager, attributes, excludeAttributes);
+        Assert.assertEquals(scimResponse.getResponseStatus(), expectedScimResponseStatus);
+    }
+
+    @DataProvider(name = "dataForTestCreateUserConflictException")
+    public Object[][] dataToTestCreatUserConflictException() {
```

Comment:
```suggestion
    public Object[][] dataToTestCreateUserConflictException() {
```

Reply:
Fixed in commit 6f5ca461fa07ac4b49182320325156a58b80b910